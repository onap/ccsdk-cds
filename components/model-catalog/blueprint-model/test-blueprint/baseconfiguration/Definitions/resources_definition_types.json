{
  "hostname": {
    "tags": "hostname",
    "name": "hostname",
    "property": {
      "description": "hostname",
      "type": "string"
    },
    "updated-by": "Singal, Kapil <ks220y@att.com>",
    "sources": {
      "input": {
        "type": "source-input"
      }
    }
  },
  "service-instance-id": {
    "tags": "service-instance-id, tosca.datatypes.Root, data_type",
    "name": "service-instance-id",
    "property": {
      "description": "To be provided",
      "type": "string"
    },
    "updated-by": "Singal, Kapil <ks220y@att.com>",
    "sources": {
      "input": {
        "type": "source-input",
        "properties": {}
      },
      "processor-db": {
        "type": "source-db",
        "properties": {
          "endpoint-selector": "dynamic-db-source",
          "query": "SELECT artifact_name FROM sdnctl.BLUEPRINT_MODEL where artifact_version=\"1.0.0\"",
          "input-key-mapping": {
          },
          "output-key-mapping": {
            "service-instance-id": "artifact_name"
          },
          "key-dependencies" : []
        }
      },
      "any-db": {
        "type": "source-db",
        "properties": {
          "endpoint-selector": "dynamic-db-source",
          "query": "SELECT artifact_name FROM sdnctl.BLUEPRINT_MODEL where artifact_version=\"1.0.0\"",
          "input-key-mapping": {
          },
          "output-key-mapping": {
            "service-instance-id": "artifact_name"
          },
          "key-dependencies" : []
        }
      }
    }
  },
  "vnf-id": {
    "tags": "vnf-id",
    "name": "vnf-id",
    "property": {
      "description": "vnf-id",
      "type": "string"
    },
    "updated-by": "Singal, Kapil <ks220y@att.com>",
    "sources": {
      "input": {
        "type": "source-input",
        "properties": {}
      }
    }
  },
  "vnf_name": {
    "tags": "vnf_name",
    "name": "vnf_name",
    "property": {
      "description": "vnf_name",
      "type": "string"
    },
    "updated-by": "Singal, Kapil <ks220y@att.com>",
    "sources": {
      "sdnc": {
        "type": "source-rest",
        "properties": {
          "verb": "GET",
          "type": "JSON",
          "url-path": "/config/GENERIC-RESOURCE-API:services/service/$service-instance-id/service-data/vnfs/vnf/$vnf-id/vnf-data/vnf-topology/vnf-parameters-data/param/vnf_name",
          "path": "/param/0/value",
          "input-key-mapping": {
            "service-instance-id": "service-instance-id",
            "vnf-id": "vnf-id"
          },
          "output-key-mapping": {
            "vnf_name": "value"
          },
          "key-dependencies": [
            "service-instance-id",
            "vnf-id"
          ]
        }
      }
    }
  },
  "aai-get-resource": {
    "tags": "aai-get",
    "name": "aai-get-resource",
    "property": {
      "description": "primary aai data to get resource",
      "type": "string"
    },
    "updated-by": "Steve, Siani <steve.djissitchi@bell.ca>",
    "sources": {
      "aai-data": {
        "type": "source-rest",
        "properties": {
          "type": "JSON",
          "verb": "GET",
          "url-path": "/aai/v22/network/generic-vnfs/generic-vnf/$vnf-id",
          "path": "",
          "input-key-mapping": {
            "vnf-id": "vnf-id"
          },
          "output-key-mapping": {
          },
          "key-dependencies": [
            "vnf-id"
          ]
        }
      }
    }
  },
  "aai-put-resource": {
    "tags": "aai-put",
    "name": "aai-put-resource",
    "property": {
      "description": "primary aai data to update resource",
      "type": "string"
    },
    "updated-by": "Steve, Siani <steve.djissitchi@bell.ca>",
    "sources": {
      "aai-data": {
        "type": "source-rest",
        "properties": {
          "type": "JSON",
          "verb": "PUT",
          "url-path": "/query?format=resource",
          "path": "",
          "payload": "{\r\n\"start\": \"\\/nodes\\/vf-modules?vf-module-name=vf-module-name\",\r\n\"query\": \"\\/query\\/related-to?startingNodeType=vf-module&relatedToNodeType=generic-vnf\"\r\n}",

          "input-key-mapping": {
            "vnf-id": "vnf-id"
          },
          "output-key-mapping": {
          },
          "key-dependencies": [
            "vnf-id"
          ]
        }
      }
    }
  },
  "aai-post-resource": {
    "tags": "aai-port",
    "name": "aai-port-resource",
    "property": {
      "description": "primary aai data to create new resource",
      "type": "string"
    },
    "updated-by": "Steve, Siani <steve.djissitchi@bell.ca>",
    "sources": {
      "aai-data": {
        "type": "source-rest",
        "properties": {
          "type": "JSON",
          "verb": "POST",
          "url-path": "/aai/add/uri/here",
          "path": "",
          "payload": "",
          "input-key-mapping": {
            "vnf-id": "vnf-id"
          },
          "output-key-mapping": {
          },
          "key-dependencies": [
            "vnf-id"
          ]
        }
      }
    }
  }
}
