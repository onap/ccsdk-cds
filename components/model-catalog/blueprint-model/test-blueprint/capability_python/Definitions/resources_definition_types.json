{
  "service-instance-id" : {
    "tags" : "service-instance-id, tosca.datatypes.Root, data_type",
    "name" : "service-instance-id",
    "property" : {
      "description" : "To be provided",
      "type" : "string"
    },
    "updated-by" : "Singal, Kapil <ks220y@att.com>",
    "sources" : {
      "input" : {
        "type" : "source-input",
        "properties" : { }
      },
      "processor-db": {
        "type": "source-db",
        "properties": {
          "endpoint-selector": "dynamic-db-source",
          "query": "SELECT artifact_name FROM sdnctl.BLUEPRINT_RUNTIME where artifact_version=\"1.0.0\"",
          "input-key-mapping": {
          },
          "output-key-mapping": {
            "service-instance-id": "artifact_name"
          },
          "key-dependencies" : []
        }
      },
      "capability": {
        "type": "source-capability",
        "properties": {
          "type": "JYTHON-COMPONENT",
          "instance-name": "SampleRAProcessor",
          "instance-dependencies": []
        }
      }
    }
  },
  "vnf-id" : {
    "tags" : "vnf-id",
    "name" : "vnf-id",
    "property" : {
      "description" : "vnf-id",
      "type" : "string"
    },
    "updated-by" : "Singal, Kapil <ks220y@att.com>",
    "sources" : {
      "input" : {
        "type" : "source-input",
        "properties" : { }
      }
    }
  }
}
