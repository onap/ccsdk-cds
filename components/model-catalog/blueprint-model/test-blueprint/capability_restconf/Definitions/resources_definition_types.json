{
  "pnf-id" : {
    "tags" : "pnf-id",
    "name" : "pnf-id",
    "property" : {
      "description" : "pnf-id",
      "type" : "string"
    },
    "updated-by" : "Rodrigo Ottero <rodrigo.ottero@est.tech>",
    "sources" : {
      "input" : {
        "type" : "source-input",
        "properties" : { }
      }
    }
  },
  "pnf-name" : {
    "tags" : "pnf-name",
    "name" : "pnf-name",
    "property" : {
      "description" : "pnf-name",
      "type" : "string"
    },
    "updated-by" : "Rodrigo Ottero <rodrigo.ottero@est.tech>",
    "sources" : {
      "input" : {
        "type" : "source-input",
        "properties" : { }
      }
    }
  }
}