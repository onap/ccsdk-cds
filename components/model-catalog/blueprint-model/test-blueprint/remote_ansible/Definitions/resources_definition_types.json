{
  "hostname" : {
    "tags" : "hostname",
    "name" : "hostname",
    "property" : {
      "description" : "hostname",
      "type" : "string"
    },
    "updated-by" : "Simard, Serge <serge@agilitae.com>",
    "sources" : {
      "input" : {
        "type" : "source-input"
      }
    }
  }
}