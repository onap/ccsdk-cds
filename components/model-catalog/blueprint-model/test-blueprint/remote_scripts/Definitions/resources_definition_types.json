{
  "hostname" : {
    "tags" : "hostname",
    "name" : "hostname",
    "property" : {
      "description" : "hostname",
      "type" : "string"
    },
    "updated-by" : "Singal, Kapil <ks220y@att.com>",
    "sources" : {
      "input" : {
        "type" : "source-input"
      }
    }
  }
}