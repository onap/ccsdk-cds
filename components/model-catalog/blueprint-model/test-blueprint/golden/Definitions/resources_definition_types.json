{
  "create_netbox_ip" : {
    "tags" : "oam-local-ipv4-address",
    "name" : "create_netbox_ip",
    "property" : {
      "description" : "netbox ip",
      "type" : "dt-netbox-ip"
    },
    "updated-by" : "adetalhouet",
    "sources" : {
      "config-data" : {
        "type" : "source-rest",
        "properties" : {
          "type" : "JSON",
          "verb" : "POST",
          "endpoint-selector" : "ipam-1",
          "url-path" : "/api/ipam/prefixes/$prefix-id/available-ips/",
          "path" : "",
          "input-key-mapping" : {
            "prefix-id" : "prefix-id"
          },
          "output-key-mapping" : {
            "address" : "address",
            "id" : "id"
          },
          "key-dependencies" : [ "prefix-id" ]
        }
      }
    }
  },
  "golden-vf-module-type" : {
    "tags" : "golden-vf-module-type",
    "name" : "golden-vf-module-type",
    "property" : {
      "description" : "golden-vf-module-type",
      "type" : "string"
    },
    "updated-by" : "adetalhouet",
    "sources" : {
      "processor-db" : {
        "type" : "source-db",
        "properties" : {
          "type" : "SQL",
          "query" : "select sdnctl.demo.value as value from sdnctl.demo where sdnctl.demo.id=:vfmoduleid",
          "input-key-mapping" : {
            "vfmoduleid" : "vf-module-number"
          },
          "output-key-mapping" : {
            "golden-vf-module-type" : "value"
          },
          "key-dependencies" : [ "vf-module-number" ]
        }
      }
    }
  },
  "hostname" : {
    "tags" : "hostname",
    "name" : "hostname",
    "property" : {
      "description" : "hostname",
      "type" : "string"
    },
    "updated-by" : "Singal, Kapil <ks220y@att.com>",
    "sources" : {
      "input" : {
        "type" : "source-input"
      }
    }
  },
  "interface-description" : {
    "tags" : "interface-description",
    "name" : "interface-description",
    "property" : {
      "description" : "interface-description",
      "type" : "string"
    },
    "updated-by" : "adetalhouet",
    "sources" : {
      "capability" : {
        "type" : "source-capability",
        "properties" : {
          "script-type" : "jython",
          "script-class-reference" : "Scripts/python/DescriptionExample.py",
          "instance-dependencies" : [ ]
        }
      }
    }
  },
  "interface-name" : {
    "tags" : "interface-name",
    "name" : "interface-name",
    "property" : {
      "description" : "interface-name",
      "type" : "string"
    },
    "updated-by" : "adetalhouet",
    "sources" : {
      "input" : {
        "type" : "source-input",
        "properties" : { }
      }
    }
  },
  "prefix-id" : {
    "tags" : "prefix-id",
    "name" : "prefix-id",
    "property" : {
      "description" : "prefix-id",
      "type" : "integer"
    },
    "updated-by" : "adetalhouet",
    "sources" : {
      "default" : {
        "type" : "source-default",
        "properties" : { }
      }
    }
  },
  "unit-number" : {
    "tags" : "unit-number",
    "name" : "unit-number",
    "property" : {
      "description" : "unit-number",
      "type" : "string"
    },
    "updated-by" : "adetalhouet",
    "sources" : {
      "input" : {
        "type" : "source-input"
      }
    }
  },
  "vf-module-number" : {
    "tags" : "vf-module-number",
    "name" : "vf-module-number",
    "property" : {
      "description" : "vf-module-number",
      "type" : "string"
    },
    "updated-by" : "adetalhouet",
    "sources" : {
      "default" : {
        "type" : "source-default",
        "properties" : { }
      }
    }
  }
}